gateway/buckets.go

```diff
@@ -165,24 +167,23 @@ func serveBucket(fs serveBucketFS) gin.HandlerFunc {
 		exists, target := fs.Exists(ctx, key, c.Request.URL.Path)
 		if exists {
 			c.Writer.WriteHeader(http.StatusOK)
-			ctype := mime.TypeByExtension(filepath.Ext(c.Request.URL.Path))
-			if ctype == "" {
-				ctype = "application/octet-stream"
-			}
-			c.Writer.Header().Set("Content-Type", ctype)
-			if err := fs.Write(ctx, key, c.Request.URL.Path, c.Writer); err != nil {
+			contentType, r, err := fs.GetPathWithContentType(ctx, key, c.Request.URL.Path)
+			if err != nil {
 				renderError(c, http.StatusInternalServerError, err)
 			} else {
+				c.Writer.Header().Set("Content-Type", contentType)
+				io.Copy(c.Writer, r)
```

Comment:
Simply call `fs.GetPathWithContentType(..)`, set that content-type and write to the response writer.